src/views/AddTimer.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useTimers } from "../context/TimerContext";
+import { nanoid } from "nanoid";
+import type { Timer } from "../types/types";
+import { StyledButtonContainer } from "../components/generic/ContainerDisplays";
+import Button from "../components/generic/Button.tsx";
+import { StyledLabel } from '../components/generic/FormStyling';
+import { FormContainer } from '../components/generic/ContainerDisplays';
+
+
+const AddTimerView = () => {
+    const { addTimer } = useTimers();
+    const navigate = useNavigate();
+
+    const [type, setType] = useState("stopwatch");
+    const [config, setConfig] = useState<any>({ hours: 0, minutes: 0, seconds: 0, numberOfRounds: 1 });
+
+    const handleTypeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        setType(event.target.value);
+        setConfig({ hours: 0, minutes: 0, seconds: 0, numberOfRounds: 1 });
+    };
+
+    const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setConfig({ ...config, [event.target.name]: Number(event.target.value) });
+    };
+
+    const handleSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        const totalTime = (config.hours * 3600 + config.minutes * 60 + config.seconds) * 1000;
+        const timerType = type as 'stopwatch' | 'countdown' | 'xy' | 'tabata';
+
+        const newTimer: Timer = {
+            id: nanoid(),
+            type: timerType,
+            config: type === "tabata"
+                ? { ...config, initialTime: config.workTime * 1000, isWorkPhase: true }
+                : { ...config, initialTime: type === "stopwatch" ? 0 : totalTime },
+            state: "notRunning",
+            timeLeft: type === "stopwatch" ? 0 : type === "tabata" ? config.workTime * 1000 : totalTime,
+        };
+        addTimer(newTimer);
+        navigate("/");
+    };
+
+
+    return (
+        <FormContainer>
+            <h2>Add New Timer</h2>
+            <form onSubmit={handleSubmit}>
+                <StyledLabel>
+                    Type:
+                    <select value={type} onChange={handleTypeChange}>
+                        <option value="stopwatch">Stopwatch</option>
+                        <option value="countdown">Countdown</option>
+                        <option value="xy">XY</option>
+                        <option value="tabata">Tabata</option>
+                    </select>
+                </StyledLabel>
+                {(type === "countdown" || type === "stopwatch") && (
```

Comment:
A lot of the logic below this is all about rendering certain input fields, depending on what type of timer it is. This makes the AddTimer component longer and more complicated than it could be. When your components start to be responsible for too many things (ex. Why should AddTimer be responsible for which inputs are presented?) I would recommend breaking out the logic into a separate component. In this case, maybe it's a `TimerInputs` or  `AddTimerForm` component.